haystack/nodes/search_engine/web.py

```diff
@@ -63,17 +63,24 @@ def run(
         labels: Optional[MultiLabel] = None,
         documents: Optional[List[Document]] = None,
         meta: Optional[dict] = None,
+        top_k: Optional[int] = None,
     ) -> Tuple[Dict, str]:
         """
         Search the search engine for the given query and return the results. Only the query parameter is used.
         :param query: The query to search for.
+        :param file_paths: Not used.
+        :param labels: Not used.
+        :param documents: Not used.
+        :param meta: Not used.
+        :param top_k: return only the top_k results.
```

Comment:
You could add that all results are returned if top_k is not set.

Reply:
> However, if I want preprocessed results, I can't choose how many pages are scraped and then preprocessed, correct

Yeah, correct. We can perhaps use top_k for that as well and limit the search results to top_k in line 198 of the `web.py`

Reply:
You make a good point. Users can now, as a workaround, init `WebRetriever` with desired `top_search_results` count but can't change that value in runtime for the number to be scraped search result. Perhaps we can leave things as is until the real need arises and then we can adjust it. LMK your thoughts.

Reply:
yes, let's leave as is 👍